### PR TITLE
Check if the package is installed before to uninstall it

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/9196

### CODE REVIEW

File: lib/tools/apk-utils.js [hunk 1]
```diff
@@ -241,21 +241,25 @@ apkUtilsMethods.waitForNotActivity = async function (pkg, act, waitMs = 20000) {
...
-    } else {
-      log.info("App was not uninstalled, maybe it wasn't on device?");
-      return false;
-    }
+    stdout = await this.adbExec(['uninstall', pkg], {timeout: 20000});
   } catch (e) {
     log.errorAndThrow(`Unable to uninstall APK. Original error: ${e.message}`);
   }
+  stdout = stdout.trim();
+  // stdout may contain warnings meaning success is not on the first line.
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: Should these warnings be outputted?

File: lib/tools/apk-utils.js [hunk 2]
```diff
@@ -241,21 +241,25 @@ apkUtilsMethods.waitForNotActivity = async function (pkg, act, waitMs = 20000) {
  */
 apkUtilsMethods.uninstallApk = async function (pkg) {
   log.debug(`Uninstalling ${pkg}`);
+  if (!this.isAppInstalled(pkg)) {
+    log.info(`${pkg} was not uninstalled, because it was not present on the device`);
+    return false;
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: Is this `false` taken as an error condition anywhere?

File: lib/tools/apk-utils.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/tools/apk-utils.js]: I do output it later if there is no Success substring there.

File: lib/tools/apk-utils.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/tools/apk-utils.js]: does this matter? These changes don't affect the previous logic of this mathod

File: lib/tools/apk-utils.js (hunk 1 above)
Comment (imurchie) [reply lib/tools/apk-utils.js]: But what if there are "warnings", whatever they are, but also success?

Comment (mykola-mokhnach) [reply lib/tools/apk-utils.js]: It's not a problem at all. Added one more log.debug call

File: lib/tools/apk-utils.js (hunk 2 above)
Comment (imurchie) [reply lib/tools/apk-utils.js]: Does it not? If the app is not installed the uninstall process doesn't get attempted. That seems like a change in the logic of this method.

I'm just wanting to make sure that when `false` is returned the calling code does not consider it to have been a problem.

Comment (mykola-mokhnach) [reply lib/tools/apk-utils.js]: The result is only used in the removeApp API call inside general.js:

```
$ cd appium-android-driver/lib/
$ grep -R 'uninstallApk' *
android-helpers.js:      await adb.uninstallApk(pkg);
commands/general.js:  return this.adb.uninstallApk(appPackage);
commands/general.js:    await this.adb.uninstallApk(this.opts.appPackage);
driver.js:      await this.adb.uninstallApk(this.opts.appPackage);
driver.js:        await this.adb.uninstallApk(this.opts.appPackage);
$ cd appium-uiautomator2-driver/lib/
$ grep -R 'uninstallApk' *
driver.js:      await this.adb.uninstallApk(this.opts.appPackage);
driver.js:          await this.adb.uninstallApk(this.opts.appPackage);
uiautomator2.js:      await this.adb.uninstallApk(apkPackage);
uiautomator2.js:      await this.adb.uninstallApk(testApkPackage);
```

Comment (mykola-mokhnach) [reply lib/tools/apk-utils.js]:
```
$ cd appium-selendroid-driver/lib/
$ grep -R 'uninstallApk' *
driver.js:      await this.adb.uninstallApk(this.opts.appPackage);
selendroid.js:      await this.adb.uninstallApk(this.modServerPkg);
```

Comment (imurchie) [reply lib/tools/apk-utils.js]: Excellent. Thanks!

**Pull request merged**